backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/LlvmModuleSpecificationImpl.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.kotlin.backend.konan
+
+import org.jetbrains.kotlin.backend.konan.descriptors.konanLibrary
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.ir.declarations.IrDeclaration
+import org.jetbrains.kotlin.ir.declarations.IrModuleFragment
+import org.jetbrains.kotlin.ir.util.module
+import org.jetbrains.kotlin.konan.library.KonanLibrary
+
+internal class LlvmModuleSpecificationImpl(private val excludedLibraries: Set<KonanLibrary>) : LlvmModuleSpecification {
```

Comment:
This `excludedLibraries` blacklisting mechanism doesn't look reliable, better to use explicit whitelisting approach.

Reply:
Why do you think so?
Any whitelisting approach would anyway be based on the "cached libraries" blacklisting mechanism, so it would be as reliable as current approach is.